### DISCUSSION TITLE
Create or generate default Grafana dashboards for the produced metrics

### DISCUSSION THREAD

**Discussion by Inithron** [quarkusio/quarkus#44165] [Design Discussions]
Hello,
a view month ago I created the service request #40933. At the end it turns out it was to big for one service request but I still think it would be a big improvement for the users if there would be an example Grafana dashboard for each [Quarkus extension which creates metrics](https://quarkus.io/guides/telemetry-micrometer#quarkus-extensions-using-micrometer). But there are now two different ways how to continue:

1. Create for each extension an own dashboard.
2. Implement a dashboard generator.

Both approaches have different advantages and disadvantages. That's why I have creates this design discussion, to discuss which approach should be implemented.

The first approach has the advantage that the dashboards can be optimized for the different extensions, for example putting the important metrics at the top, use different sizes of the panels, grouping of related metrics, use special graphs for special metrics, etc.
One disadvantage is to find contributors who are familiar with the metrics produced by the different extensions and the creation of dashboards. So it could take a while until an example board for each extension is available. Another one is that these dashboards need manual maintenance if an extension creates new metrics or existing ones are renamed.

For the second approach this [Grafana Dashboard Generator](https://github.com/jamesfalkner/microprofile-grafana) could be used as template or inspiration.
Disadvantages:
1. The mentioned dashboard generator is no more maintained and not working out of the box with the metrics produced by Quarkus.
2. So this generator needs to be adapted or a new one needs to be implemented.
3. By default one dashboard is created which contains all metrics. But this is confusing / overwhelming the user. So until now it is unclear how to create one dashboard per extension.

After a dashboard generator was implemented, it has the following advantages:
1. Supporting of all existing and future extensions which produce metrics.
2. The generated dashboards are always up to date.
4. Custom metrics are supported out of the box.

**Comment by brunobat** [comment]
CC @melloware, @alesj, @jmartisk

**Reply by melloware** [reply]
1. Generator sounds great but really difficult to implement properly.
2. I would prefer people just start donating a dashboard per extension like...
- quarkus-kafka.json
- quarkus-jdbc.json
- etc etc
